Software_Carbon_Intensity/Appendix_A.md

```diff
@@ -1 +1,100 @@
+
+From an SRE principles and alignment perspective then, organizations would like to track carbon emissions from their applications using the SCI score and raise alerts on breach, i.e. if it increases beyond a certain percentage . If an application has a SCI score of x, organizations would then track variation from this value and configure monitoring thresholds.
+
+**How would you then correctly define the metric as per SRE principles?**
+
+In the above formula for SCI, the baseline is a key aspect. We will explain with an example where we are considering the baseline as "one instance of a batch job".  The batch job is a piece of component within a larger "software system" or an "application" which could potentially be a web application workload with a batch job doing a long running business process that does not need user interactions.
+
+Let us assume that the SCI value of a batch job running in West Europe has a CI value of 100 kgCO2 per instance of Azure WebJob. By an initial assumption, let us assume that the SLO for SCI has been defined as not more than 20% variation. If during the operating window of the job, the service cranks up and the carbon intensity increases to 121 KgCO2 , then an alert has to be signalled. However this is theoretical.  We have to look at this increase in the context of many factors like how the interplay of SCI increases with other SLOs like latency, performance, etc; how the Western European datacenter was powered (% of coal vs. renewables) during the time of heightened operation of this WebJob; inefficient threading and garbage collection practices that could have been in the code that surfaced during the peak operation, etc.  
+
+When this incident (as per SRE principles, this is an incident that should be monitored and alerted like a Sev 1,2,3 incident) happens, there could be multiple tuning techniques that might be used to address this. One tuning technique to address this is to try moving the workload to a different datacenter that is powered more by renewables (by calling the WattTime API) or shifting the workload to a different time of the day when the energy input mix is greener. These are techniques that need detailed and vetted data upfront for the "orchestration algorithm" to make dynamic decisions regarding how to move the workload. However, right now, we do not have defined and foolproof information to tell how much do each of these tuning techniques will contribute to managing the increased carbon intensity. This data has to be collated and cross verified over a long period of time going forward to come up with authentic deductions. 
+
+Hence for the initial version of the specification, we can raise the level of abstraction for monitoring of SCI at an application level rather than at the individual component level. i.e we will keep the baseline for software carbon intensity at the "application level" rather than a batch job, ML job, API call, etc. 
```

Comment:
nit: formatting

```suggestion
Hence for the initial version of the specification, we can raise the level of abstraction for monitoring of SCI at an application level rather than at the individual component level. i.e. we will keep the baseline for software carbon intensity at the "application level" rather than a batch job, ML job, API call, etc. 
```